tools/hack/golang/update

```diff
@@ -5,6 +5,7 @@ set -o nounset
 set -o pipefail
 
 src="$(cd "$(dirname "${BASH_SOURCE[0]}")" && pwd)"
+TMPDIR=$(mktemp -d)
```

Comment:
@smarterclayton is this something that is set on a Mac by default?

Reply:
This is a mac thing. I'll try to get around to rewriting this knowing that variable could possibly be defined.